packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -121,15 +120,22 @@ export class RunningSummarizer implements IDisposable {
             },
         );
 
-        if (!disableHeuristics) {
+        if (configuration.state !== "disableHeuristics") {
+            assert(this.configuration.state === "enabled", "Configuration state should be enabled");
+            const heuristics: ISummaryConfigurationHeuristicSettings = {
+                    ...this.configuration,
+            };
             this.heuristicRunner = new SummarizeHeuristicRunner(
                 heuristicData,
-                configuration,
+                heuristics,
                 (reason) => this.trySummarize(reason));
         }
 
+        assert (this.configuration.state !== "disabled", "Summary not supported with configuration disabled");
```

Comment:
It's better for these kinds of checks to be at compile time, not at runtime.
I.e. 
private readonly configuration: ISummaryConfiguration & {state : "enabled" | state: "disableHeuristics" },

Or whatever change required (if compiler does not like that) to ensure that requirement is pushed up the stack and is validated at compile time.

Reply:
The problem is that if we do not have the assert, we have problems accessing the fields 
it will complain about this.configuration.maxAckWaitTime
not existing when state == disabled.  

Thoughts ?